core/commonMain/src/kotlinx/serialization/encoding/Decoding.kt

```diff
@@ -543,14 +496,22 @@ public interface CompositeDecoder {
      * Implementation can safely ignore it and return a new value, efficiently using 'the last one wins' strategy,
      * or apply format-specific aggregating strategies, e.g. appending scattered Protobuf lists to a single one.
      */
-    @Suppress("DEPRECATION_ERROR")
     @ExperimentalSerializationApi
     public fun <T : Any> decodeNullableSerializableElement(
         descriptor: SerialDescriptor,
         index: Int,
         deserializer: DeserializationStrategy<T?>,
         previousValue: T? = null
-    ): T? = decodeNullableSerializableElement(descriptor, i = index, deserializer = deserializer)
+    ): T?
+
+    @Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE", "DeprecatedCallableAddReplaceWith")
+    @kotlin.internal.LowPriorityInOverloadResolution
+    @Deprecated(decodeMethodDeprecated, level = DeprecationLevel.ERROR)
+    public fun <T : Any?> decodeSerializableElement(
+        descriptor: SerialDescriptor,
+        i: Int, // renamed from index to be called even with LowPriority
+        deserializer: DeserializationStrategy<T>
+    ): T = decodeSerializableElement(descriptor, i, deserializer, null)
 
     @Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE", "DeprecatedCallableAddReplaceWith")
     @kotlin.internal.LowPriorityInOverloadResolution
```

Comment:
Compiler _still_ uses these two deprecated declarations. It should be fixed in 1.4.20

Reply:
Can we deprecate them as hidden though?

Reply:
Actually, I've checked plugin code and it seems to me that plugin uses 4-argument functions. How you determined that these are still in use?

Reply:
If I remove them, tests starts to fail with NSME

Reply:
Ok, I've figured it out: compiler generates wrong code only for JVM and only for code path where `decodeSequentially = true`. Seems I was using there 3-arg function because, well, `oldValue` does not make sense for sequential reading. So we indeed have to leave this functions hidden if we want sequential reading working (for me it looks like a better solution instead of breaking it)